### ISSUE TITLE
Read part of large ome.tiff image without zarr

### ISSUE THREAD

**Issue by LionelArn2** [cgohlke/tifffile#213]
Hello @cgohlke,

I'm trying to extract areas of large (100GB-1TB) multi-page pyramidal images (with different compression method) without loading the entire image into memory, and for that the zarr [example](https://github.com/cgohlke/tifffile/blob/08290c7ae44a4248f64d7cd0e74361f570f56e79/tifffile/tifffile.py#L654-L672) you provided works quite well 👍 thanks for that.

This involves the usage of zarr however, and I'm trying to minimize as much as possible the amount of dependencies in a large project. I could see an [example](https://gist.github.com/rfezzani/b4b8852c5a48a901c1e94e09feb34743) online without zarr which works OK in certain conditions but typically behaves poorly in multithreading (need to lock the entire crop method).

Do you think it would it be complex to achieve this without zarr?

**Comment by cgohlke** [comment]
I think Zarr is basically the community standard for accessing chunked, compressed, n-dimensional arrays. Using a Zarr store enables seamless integration with powerful tools, such as dask. It is unfortunate that Zarr has a significant latency overhead. Zarr's dependencies are actually not too large, but it would be nice to use it without numcodecs, which is actually not required by tifffile/ZarrTiffStore.

Some libraries and tools (e.g., `opentile` and the example you linked) implemented other APIs (e.g., OpenSlide) on top of TiffFile. A potential issue is, as you noted, that the implementation might only work for specific TIFF formats.

A more general approach to read parts of a large TIFF without Zarr should be to use the `ZarrTiffStore` returned by tifffile and implement a "mini zarr" to get (by indexing the store) the `shape`, `chunk shape`, `dtype` array properties, and all the chunks that are part of a region, copying the chunks into a contiguous numpy array. The `ZarrTiffStore` itself does not depend on Zarr or numcodecs...

**Comment by LionelArn2** [comment]
Thanks for the quick answer. I made a small edit to my question before seeing that you had already answered, sorry about that.

In the end the example which directly uses the file handle seems to work fine as long as you do not allow multiple threads to do it at the same time (or even access the tiff page). As you mentioned, I can see that this is faster than Zarr since we avoid its overhead. I guess I could go for a simplistic Zarr implementation like you suggested but since I did not run into any issue when properly locking the sensitive part of code, I'll keep the direct "file handle" solution for now.

Thanks for your help and for developing Tifffile, it's quite a complete solution!

Lionel
